desc/profiles.py

```diff
@@ -885,8 +885,12 @@ def compute(self, params=None, grid=None, dr=0, dt=0, dz=0):
             return jnp.zeros_like(xq)
         x = self._knots
         f = params
-        df = self._Dx @ f
-        fq = interp1d(xq, x, f, method=self._method, derivative=dr, extrap=True, df=df)
+        if self._method == "monotonic":
```

Comment:
I think it might actually be faster in many cases to just not precompute `df`:

```python
import numpy as np
from jax import jit
from desc.interpolate import interp1d, _approx_df
from desc.profiles import PowerSeriesProfile
from desc.grid import ConcentricGrid
p = PowerSeriesProfile([1,0,-2,0,1])
s = p.to_spline()
grid = ConcentricGrid(24,24,24)
s.grid = grid
p.grid = grid

Dx = _approx_df(
    s._knots, np.eye(s._knots.size), s._method, axis=0
    )

def interp(f):
    return interp1d(
        grid.nodes[:,0], 
        s._knots, 
        f, 
        method=s._method, 
        derivative=0, 
        extrap=True,
       )

def dinterp(f):
    fx = Dx @ f
    return interp1d(
        grid.nodes[:,0], 
        s._knots, 
        f, 
        method=s._method, 
        derivative=0,
        extrap=True, 
        fx=fx
    )

%timeit _ = jit(interp)(s.params).block_until_ready()
%timeit _ = jit(dinterp)(s.params).block_until_ready()
```

without precompute:
```1.03 ms ± 270 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)```
with precompute:
```1.24 ms ± 296 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)```

Computing df with the matrix multiply takes `O(n^2)` operations (since its basically doing it for any possible values of `f`), but in theory calculating it on the fly only requires `O(n)` operations

Reply:
hm I should have profiled that... would have saved me a lot of time this morning lol. Thanks, I will change that, I will keep the changes to the monotonic spline stuff though if it could be useful for the higher dimensional interpolation (I should add it as a method for those too)